File: chefstlt.java
/*Write a program which will take two string and find the maximum and 
minimum difference between them so that they will be similar*/
import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;
class chefstlt 
{
	public static void main(String[] args) throws IOException 
	{
      BufferedReader inp=new BufferedReader(new InputStreamReader(System.in));
      String s1,s2;
      int testCases,min,max;
      testCases=Integer.parseInt(inp.readLine());
      while(testCases>0)
      {
      	s1=inp.readLine();
      	s2=inp.readLine();
        min=minumum(s1,s2);
        max=maximum(s1,s2);
        System.out.printf("%d %d\n",min,max);
      }
	}

	public static int minimum(String s1,String s2)
	{
        
	}

	public static int maximum(String s1,String s2)
	{

	}
}

File: chefstlt.cpp
//This program will solve the chef problem for the sttring
#include<iostream>
using namespace std;
#include<string>
int maximum(string,string);
int minimum(string, string);
int main()
{
	string input1,input2;
	int testCases;
	cin>>testCases;
	while(testCases--)
	{
		cin>>input1>>input2;
		cout<<minimum(input1,input2)<<" ";
		cout<<maximum(input1,input2)<<endl;
	}
	return 0;
}
int minimum(string s1,string s2)
{
	int min=0;
    for(int i=0;i<s1.length();i++)
    {
    	if(s1[i]!=s2[i] && s1[i]!='?' && s2[i]!='?')
    		min++;
    }
    return min;
	
}

int maximum(string s1,string s2)
{
   int max=0;
   for(int i=0;i<s1.length();i++)
   {
   	if(s1[i]!=s2[i] || s1[i]=='?' || s2[i]=='?')
   		max++;
   }
   return max;
}